pydantic/fields.py

```diff
@@ -153,6 +153,20 @@ def get_constraints(self) -> Set[str]:
         """
         return {attr for attr, default in self.__field_constraints__.items() if getattr(self, attr) != default}
 
+    def update_from_config(self, from_config: Dict[str, Any]) -> None:
+        """
+        Update this FieldInfo based on a dict from get_field_info, only fields which have not been set are dated.
+        """
+        for attr_name, value in from_config.items():
+            try:
+                current_value = getattr(self, attr_name)
+            except AttributeError:
+                # attr_name is not an attribute of FieldInfo, it should therefore be added to extra
+                self.extra[attr_name] = value
+            else:
+                if current_value is None:
```

Comment:
```suggestion
                if current_value is self.__field_constraints__.get(attr_name, None):
```
I wonder if we should do this to support 
```py
from pydantic import BaseModel, Field


class Foo(BaseModel):
    a: str = Field(...)

    class Config:
        validate_assignment = True
        fields = {'a': {'allow_mutation': False}}

f = Foo(a='x')
f.a = 'y'  # should fail
```

Reply:
LOL 😄 just saw your last commit

Reply:
ha, I was just writing a comment about `allow_mutation`.